### PR TITLE
Add Uhuru RAVEN and NUCLEO F767ZI support

### PR DESCRIPTION
**Pull request by toyowata**
### Adding new target

[Uhuru Raven](https://os.mbed.com/platforms/UH-ENE01/)

Internal Flash : 2MB
RAM : 512kB
Connectivity : Wi-Fi (ESP32)

kvstore : internal flash
Update storage : internal flash

[NUCLEO F767ZI](https://os.mbed.com/platforms/ST-Nucleo-F767ZI/)

Internal Flash : 2MB
RAM : 512kB
Connectivity : Ethernet

kvstore : internal flash
Update storage : internal flash

### Summary of changes 

[x] I confirm this contribution is my own and I agree to license it with Apache 2.0.

[x] I confirm the moderators may change the PR before merging it in.

For new board enablements only:

[x] I confirm the board is [Mbed Enabled](https://www.mbed.com/en/about-mbed/mbed-enabled/introduction/) and passes the Mbed Enabled test set.

[] I confirm the contribution has been tested properly and the tests results for [TESTS](https://github.com/ARMmbed/mbed-os-example-pelion/tree/master/TESTS) are attached.

**The test can not run on this target.  See https://github.com/ARMmbed/mbed-os-example-pelion/issues/109 for more detail.**  I did manually test connect and update features for this target and works fine.

[x] I confirm `mbed-os.lib` and `mbed-cloud-client.lib` hashes or the content in folders `mbed-os` and `mbed-cloud-client` were not modified in order to pass the tests.

- Maintainers of this repository update the Mbed OS and Client hashes.
- Please report any issues through issues in relevant repositories.

cc @MarceloSalazar

### CODE REVIEW

**Comment by iot-pdmc** [comment]
Thanks for your changes.
@ARMmbed/pdmc-integrators please review.

**Comment by MarceloSalazar** [comment]
Thanks Toyo!
@teetak01 FYI these two targets have the same MCU: F767ZI

File: mbed_app.json [hunk 1]
```diff
@@ -428,6 +428,53 @@
...
+            "update-client.application-details"         : "(MBED_CONF_STORAGE_TDB_INTERNAL_INTERNAL_BASE_ADDRESS + MBED_CONF_STORAGE_TDB_INTERNAL_INTERNAL_SIZE)",
+            "update-client.storage-address"             : "(MBED_ROM_START + (MBED_ROM_SIZE/2))",
+            "update-client.storage-size"                : "((MBED_ROM_SIZE/2) - MBED_CONF_STORAGE_TDB_INTERNAL_INTERNAL_SIZE)",
+            "update-client.storage-locations"           : 1,
+            "update-client.storage-page"                : 1,
+            "update-client.firmware-header-version"     : "2",
+            "mbed-cloud-client.update-storage"          : "ARM_UCP_FLASHIAP",
+            "storage.storage_type"                      : "TDB_INTERNAL",
+            "storage_tdb_internal.internal_base_address": "(MBED_ROM_START + 32*1024)",
+            "storage_tdb_internal.internal_size"        : "(32*2*1024)"
```
Comment (teetak01) [inline mbed_app.json]: Any reason why the TDB internal is 64 KiB and not 32 KiB?

File: mbed_app.json [hunk 2]
```diff
@@ -428,6 +428,53 @@
...
+            "update-client.bootloader-details"          : "0x8005d30",
+            "update-client.application-details"         : "(MBED_CONF_STORAGE_TDB_INTERNAL_INTERNAL_BASE_ADDRESS + MBED_CONF_STORAGE_TDB_INTERNAL_INTERNAL_SIZE)",
+            "update-client.storage-address"             : "(MBED_ROM_START + (MBED_ROM_SIZE/2))",
+            "update-client.storage-size"                : "((MBED_ROM_SIZE/2) - MBED_CONF_STORAGE_TDB_INTERNAL_INTERNAL_SIZE)",
+            "update-client.storage-locations"           : 1,
+            "update-client.storage-page"                : 1,
+            "update-client.firmware-header-version"     : "2",
+            "mbed-cloud-client.update-storage"          : "ARM_UCP_FLASHIAP",
+            "storage.storage_type"                      : "TDB_INTERNAL",
+            "storage_tdb_internal.internal_base_address": "(MBED_ROM_START + 32*1024)",
```
Comment (teetak01) [inline mbed_app.json]: This probably should be in (MBED_ROM_START + (MBED_ROM_SIZE/2) + MBED_CONF_STORAGE_TDB_INTERNAL_INTERNAL_SIZE) ?

File: mbed_app.json (hunk 1 above)
Comment (toyowata) [reply mbed_app.json]: Because minimum size of the STM32F767 flash sector is 32kb.

**Comment by toyowata** [comment]
@teetak01 

STM32F767 flash memory block info:
stm32f767_flash

If I follow the recommendation config [here](https://www.pelion.com/docs/device-management/current/porting/porting-a-new-target-for-mbed-os-using-internal-flash-for-secure-storage-kvstore.html), kvstore is located to Sector 8 and 9 (256+256 KB), which is not optimum config.

So, I defined flash configuration as below:

|Sector #| Sector size|Description|
|---|---|---|
|Sector 0|32 KB|Bootloader|
|Sector 1|32 KB|kvstore 0|
|Sector 2|32 KB|kvstore 1|
|Sector 3|32 KB|Application|
|Sector 4|128 KB|Application|
|Sector 5|256 KB|Application|
|Sector 6|256 KB|Application|
|Sector 7|256 KB|Application|
|Sector 8|256 KB|Update storage|
|Sector 9|256 KB|Update storage|
|Sector 10|256 KB|Update storage|
|Sector 11|256 KB|Update storage|

**Comment by teetak01** [comment]
build-only

**Comment by toyowata** [comment]
I just got test result from device using latest release of pelion-e2e-python-test-library:

```
----------------------------------------------------- live log sessionfinish ------------------------------------------------------
2020-02-06 16:58:24.971      INFO   -----  TEST RESULTS SUMMARY  -----
2020-02-06 16:58:24.972      INFO   [ check the complete fail reasons and code locations from this log or html report ]
2020-02-06 16:58:24.972      INFO   [passed] - dev-client-tests.py::test_01_get_device_id - (0.790s)
2020-02-06 16:58:24.973      INFO   [passed] - dev-client-tests.py::test_02_put_resource - (10.753s)
2020-02-06 16:58:24.973      INFO   [passed] - dev-client-tests.py::test_03_get_resource - (1.756s)
2020-02-06 16:58:24.973      INFO   [passed] - dev-client-tests.py::test_04_subscribe_resource - (6.444s)
2020-02-06 16:58:24.973      INFO   [passed] - dev-client-tests.py::test_05_factory_reset - (20.861s)
```

**Comment by iot-pdmc** [comment]
# [Test run: SUCCESS](https://jenkins-client.isgtesting.com/job/mbed-os-example-pelion/593/)
***
[Build artifacts](http://pdmc-ci.s3-website-eu-west-1.amazonaws.com/?prefix=jenkins-ci/mbed-os-example-pelion/593)

posted-by-tag: https://jenkins-client.isgtesting.com/job/mbed-os-example-pelion/

**Pull request merged**